RIOT-OS/RIOT
# boards/nrf52*dk: Enable Pinreset after Flashing

### Contribution description

As outlined in #20775, there was a silicon change in the nRF52 chips at some revision making the UICR register non-persistent. This means that the applied configuration is deleted when the flash is erased and programmed.
The reset pin is configured in this UICR register as well, so after flashing RIOT to (any) nRF52 chip without a bootloader, the reset pin won't work anymore.

This PR adds a JLINK_POST_FLASH command which configures the UICR->PSELRESET registers while the nRF52 chip is still unlocked from Flashing.

The JLINK_POST_FLASH variable is added to the builsystem sanity check as well, so it is evaluated alongside JLINK_PRE_FLASH. I'm not sure if that was required, but now it is consistent.

Request for comments: The documentation should be updated accordingly and the old tool in `dist/tools/nrf52_resetpin_cfg` is obsolete and could be deleted. This PR works with all silicon revisions of the nRF52 and the tool does not.
Should I do that in this PR or do a separate PR?




### Testing procedure

You can use any application you want and flash it to either a nRF52DK board (with the nRF52832DK) or an nRF52840DK from Nordic and press the Reset button.
The board should reset.

```
BOARD=nrf52dk make -C examples/default flash term
```




### Issues/PRs references

This fixes #20775.
